geode-core/src/main/java/org/apache/geode/internal/cache/wan/serial/ConcurrentSerialGatewaySenderEventProcessor.java

```diff
@@ -374,6 +380,16 @@ public void initializeEventDispatcher() {
 
   }
 
+  @Override
+  protected void registerEventDroppedInPrimaryQueue(EntryEventImpl droppedEvent) {
+    this.getSender().setModifiedEventId(droppedEvent);
```

Comment:
I was thinking you'd call the sender's setModifiedEventId in the sender right after you create the EntryEventImpl clone, but since its not necessary in the parallel case, I guess its ok to do it here. If you did need to set it in the parallel case, it might be better to move this up to the AGS.